src/tests/test_clang_bug.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+* (C) 2020 Jack Lloyd, René Meusel, Hannes Rantzsch
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+/*
+ * This test case ensures that we avoid crashing in PKCS8::load_key due to a bug in Clang 8.
+ *
+ * A detailed description of the issue can be found here: https://github.com/randombit/botan/issues/2255.
+ * In short, Clang 8 performs a double-free on captured objects if an exception is thrown within a lambda. This case can
+ * occur when PKCS8::load_key fails to load the key due to a wrong password. The password needs to be long enough to not
+ * be small buffer optimized.
+ *
+ * The Clang bug is tracked here and apparently won't be fixed in Clang 8: https://bugs.llvm.org/show_bug.cgi?id=41810.
+ * Other Clang versions seem not to be affected.
+ */
+
+#include "tests.h"
+
+#if defined(BOTAN_HAS_PUBLIC_KEY_CRYPTO)
+   #include <botan/system_rng.h>
+   #include <botan/pkcs8.h>
+   #include <botan/data_src.h>
+#endif // BOTAN_HAS_PUBLIC_KEY_CRYPTO
+
+namespace Botan_Tests {
+
+#if defined(BOTAN_HAS_PUBLIC_KEY_CRYPTO)
+
+namespace {
+
+// The key needs to be a PKCS #8 encoded and encrypted private key.
+std::string getPrivateKey()
+   {
+   return
+      "-----BEGIN ENCRYPTED PRIVATE KEY-----"
```

Comment:
For future reference: Do you remember the correct password? The test itself must use a wrong password, you mention that in the long description above. Future-me would find it helpful to see the "correct" password as another hint when quickly reading this test case